domain_content/domain_content.module

```diff
@@ -33,3 +33,37 @@ function domain_content_domain_operations(DomainInterface $domain, AccountInterf
 
   return $operations;
 }
+
+/**
+ * Implements hook_requirements.
+ */
+function domain_content_requirements($phase) {
+  $requirements = array();
+  $allow = TRUE;
+  if ($phase == 'install') {
+    $list['user'] = 'user';
+    $node_types = \Drupal::entityTypeManager()->getStorage('node_type')->loadMultiple();
+    foreach ($node_types as $type => $info) {
+      $list[$type] = 'node';
+    }
+    // Check for required fields.
+    foreach ($list as $bundle => $entity_type) {
+      $id = $entity_type . '.' . $bundle . '.' . DOMAIN_ACCESS_FIELD;
+      if (!$field = \Drupal::entityTypeManager()->getStorage('field_config')->load($id)) {
+        $allow = FALSE;
```

Comment:
Maybe break early? if there is at least one FALSE, there is no need to check all of them